### DISCUSSION TITLE
Is the IntelliSense code linter for C++ supposed to work?

### DISCUSSION THREAD

**Discussion by sweemer** [microsoft/vscode-cpptools#11861] [Ideas]
Hello, I am wondering whether Visual Studio's [IntelliSense code linter for C++](https://learn.microsoft.com/en-us/cpp/ide/cpp-linter-overview?view=msvc-170) is supposed to work in Visual Studio Code. If so, then it does not seem to work out of the box for me, so how can I get it to work? If not, then can this feature be added to Visual Studio Code as well?

**Comment by bobbrow** [comment]
No, we have not added the Visual Studio linter rules into the C++ extension for VS Code yet.  However, we do have built in support for clang-tidy if you'd like to give that a try.  Start with the `C_Cpp.codeAnalysis.clangTidy.enabled` setting and if you want to make any changes to the default experience there are other settings with the `C_Cpp.codeAnalysis.clangTidy` prefix that you can experiment with.

A quick search doesn't turn up a feature request for the Visual Studio linters in our issues list, so you are welcome to open a new issue to request it.

**Reply by sweemer** [reply]
I agree that the built-in support for clang-tidy is useful. It's my impression that the IntelliSense linter is a bit faster and lighter-weight than clang-tidy, so I might prefer it over clang-tidy for daily use.
